Refactoring nuage templates

sunny-verma opened the pull request.

sunny-verma committed:
PROD-10137: Refactoring nuage heat templates

This patch includes:

  1. Moving Nuage VRS/AVRS templates to /home/stack
  2. Removing obsolete code from patching script which is not required as of #1
  3. Support for multirole AVRS starting from 5.4.1u4
  4. Documentation changes:
	* Accommodating above changes
	* PROD-9702: Adding RHSB support
	* Separating 5.4.1u4 document with earlier version
  5. Using Signed packages support for Under cloud patching and Overcloud Deployment
  6. PROD-10132: Solving Issues on github.com
	* issues/68
	* issues/63
	* issues/51
  7. removing unwanted dependencies like python-novaclient ( as python2-novaclient is already present in the base image)

sunny-verma commented on nuage-tripleo-heat-templates/scripts/create_roles/create_compute_avrs_multirole_sample.sh:
```diff
@@ -0,0 +1,30 @@
+roles=(ComputeAvrsSingle ComputeAvrsDual)
+echo "creating ${roles[*]} Role"
+
+cp -r /home/stack/openstack-tripleo-heat-templates/roles/* ../../roles/
```
change to /usr/share/

sunny-verma commented on nuage-tripleo-heat-templates/scripts/create_roles/create_compute_avrs_role.sh:
```diff
@@ -0,0 +1,23 @@
+source /home/stack/stackrc
+
+echo "creating ComputeAvrs Role"
+cp -r /home/stack/openstack-tripleo-heat-templates/roles/* ../../roles/
```
change to /usr/share/

psairam commented on Documentation/5.4.1/README.rst:
```diff
@@ -165,19 +165,18 @@ The integration includes the following steps:
-    - For AVRS integration, see the `Sample Templates`_ section for some probable values of the parameters in the ``avrs-environment.yaml`` file and we also need to create a new AVRS role similar to the upstream Compute role.
+    - For AVRS integration, some of the parameters in ``fast-path.env`` needs to be configured in the Heat templates. We use ``compute-avrs-environment.yaml`` environment file to configure these values.
+    - For AVRS integration, see the `Sample Templates`_ section for some probable values of the parameters in the ``compute-avrs-environment.yaml`` file and we also need to create a new AVRS role similar to the upstream Compute role.
+    - For AVRS intergration, we can also create Multiple roles which we can pass different sets of configuration on those AVRS Compute Node. Sample enviroment file can be found `here <../../nuage-tripleo-heat-templates/environments/avrs-multirole-environment-sample.yaml>`_
```
This link says, "Page not found", because env file name is "compute-avrs-multirole-environment-sample.yaml"

psairam commented on Documentation/5.4.1/README.rst:
```diff
@@ -312,15 +321,46 @@ Follow the instructions in `README.md
+
+.. Note:: Above kernel arguments are consumed by the another env file which include in deployment command `/usr/share/openstack-tripleo-heat-templates/environments/host-config-and-reboot.yaml`
+
+2. Use the `create_compute_avrs_role.sh <../../scripts/create_roles/create_compute_avrs_role.sh>`_ to create a roles file called ``compute-avrs-role.yaml``.
```
the path in the link should be "../../nuage-tripleo-heat-templates/scripts/create_roles/create_compute_avrs_role.sh", it is missing "nuage-tripleo-heat-templates"

psairam commented on Documentation/5.4.1/README.rst:
```diff
@@ -312,15 +321,46 @@ Follow the instructions in `README.md
 
-Above command will create a new ``ComputeAvrs``  role for your deployment, and compare it with sample avrs-role.yaml provided at https://github.com/nuagenetworks/nuage-ospdirector/tree/OSPD13/avrs/avrs-role.yaml .
-(Please Note, given ``avrs-role.yaml`` file can get updated with newer release )
+Above command will create a new ``ComputeAvrs``  role for your deployment, and compare it with sample compute-avrs-role-sample.yaml provided at https://github.com/nuagenetworks/nuage-ospdirector/tree/OSPD13/nuage-tripleo-heat-templates/roles/compute-avrs-role-sample.yaml .
```
lets use relative path, rather than absolute path

psairam commented on Documentation/5.4.1/README.rst:
```diff
@@ -312,15 +321,46 @@ Follow the instructions in `README.md
+Above command will create a new ``ComputeAvrs``  role for your deployment, and compare it with sample compute-avrs-role-sample.yaml provided at https://github.com/nuagenetworks/nuage-ospdirector/tree/OSPD13/nuage-tripleo-heat-templates/roles/compute-avrs-role-sample.yaml .
+.. Note:: given ``compute-avrs-role.yaml`` file can get updated with newer release )
+
+For the users that what to deploy mutli-role for the AVRS. We are providing sample `environment <../../environments/compute-avrs-mutlirole-environment-sample.yaml>`_ file and automated `scrip <../../scripts/create_compute_avrs_multirole_sample.sh>`_ to create ComputeAvrsSingle & ComputeAvrsDual role. You can edit these files with your requirements to create new roles.
```
the relative path links are missing "nuage-tripleo-heat-templates" in them. Also, typo in name of "scrip" it should be "scripts".

psairam commented on Documentation/5.4.1/README.rst:
```diff
@@ -312,15 +321,46 @@ Follow the instructions in `README.md
+
+::
+
+    Environment file is at: ../../environments/compute-avrs-mutlirole-environment-sample.yaml
```
lets put point 3 at point 1

psairam commented on Documentation/5.4.1/README.rst:
```diff
@@ -407,15 +447,33 @@ For AVRS integration, follow these steps:
     openstack flavor create --id auto --ram 4096 --disk 40 --vcpus 1 computeavrs
     openstack flavor set --property "cpu_arch"="x86_64" --property "capabilities:boot_option"="local" --property "capabilities:profile"="computeavrs" computeavrs
 
+For multi-roles, ( assuming you want to create two different flavor for different roles )
```
remove assuming and add "for single role" and "for multi roles"

psairam commented on Documentation/5.4.1/README.rst:
```diff
@@ -426,6 +484,20 @@ For AVRS integration, follow these steps:
     OvercloudComputeAvrsFlavor: computeavrs
     ComputeAvrsCount: 2
 
+For multi-roles, set the `node-info.yaml` with the corresponding role name. The following example shows how to create a deployment with one Controller node, two Compute nodes and two different multi-role Avrs nodes:
```
instead of `two different multi-role Avrs` lets say one ComputeAvrsSingle and one ComputeAvrsDual.

psairam commented on Documentation/5.4.1/README.rst:
```diff
@@ -438,11 +510,11 @@ For AVRS integration, follow these steps:
 ::
 
-    openstack overcloud roles generate Controller Compute ComputeSriov -o /home/stack/templates/sriov-role.yaml
+    openstack overcloud roles generate Controller Compute ComputeSriov -o /home/stack/nuage-tripleo-heat-templates/roles/compute-compute-sriov-role.yaml
```
typo in compute-compute-sriov-role.yaml

psairam commented on Documentation/5.4.1/README.rst:
```diff
@@ -689,7 +761,7 @@ In OSPD 13 and later, /usr/share/openstack-tripleo-heat-templates/environments/n
     ::
 
-        openstack overcloud roles generate Controller Compute -o /home/stack/templates/ironic-role.yaml
+        openstack overcloud roles generate Controller Compute -o /home/stack/nuage-tripleo-heat-templates/roles/compute-ironic-role.yaml
```
ironic will be enabled on controllers right, so having compute in the name will be appropriate?

psairam commented on image-patching/stopgap-script/nuage_overcloud_full_patch.py:
```diff
@@ -29,8 +29,7 @@
 # 5. Install VRS
 # 6. Unsubscribe from RHEL
 # 7. Add the files post-patching
```
lets remove this line as it is no longer valid

psairam commented on Documentation/5.4.1/README.rst:
```diff
@@ -273,30 +272,40 @@ Follow the steps in https://access.redhat.com/documentation/en-us/red_hat_openst
-1. Create an environment file called avrs-environment.yaml in usr/share/openstack-tripleo-heat-templates/environments/.
+1. Edit an environment file called compute-avrs-environment.yaml in /home/stack/nuage-tripleo-heat-templates/environments/.
+
+a. For configuring  `/etc/fast-path.env` we set the following parameters:
```
in rendered format "a" is shown as "1", please fix it

psairam commented on Documentation/5.4.1/README.rst:
```diff
@@ -11,15 +11,7 @@
 .. ..  =========  =======    =========
```
change the directory structure

psairam commented on Documentation/5.4.1/BEFORE_5.4.1u4/README.rst:
```diff
@@ -0,0 +1,1762 @@
+    FastPathOptions        =====>    FP_OPTIONS
+
+
+b. For configuring `nova.conf` we set the following parameters:
```
modify this by saying, 6wind or Avrs requires to configure monkey_patch parameters in nova.conf and we use below to configure them.

psairam commented on Documentation/5.4.1/BEFORE_5.4.1u4/README.rst:
```diff
@@ -0,0 +1,1762 @@
+
+1. Edit an environment file called compute-avrs-environment.yaml in /home/stack/nuage-tripleo-heat-templates/environments/.
+
+a. For configuring  `/etc/fast-path.env` we set the following parameters:
```
change this as, "Virtual Accelerator requires information like which logical cores run the fast path, list of ports enabled in the fast path, additional fast path options etc.., to be set inside /etc/fast-path.env. Below is the mapping between parameters in heat template to parameters in fast-path.env".
Also, lets define these params in Parameters default section.

psairam commented on Documentation/5.4.1/BEFORE_5.4.1u4/README.rst:
```diff
@@ -0,0 +1,1762 @@
+          monkey_patch_modules:
+             value: nova.virt.libvirt.vif:openstack_6wind_extensions.queens.nova.virt.libvirt.vif.decorator
+
+c. For configuring kenerl arguments we set following parameters:
```
change this as, "Virtual Accelerator requires hugepages to be configured and the value can be varied. We also need to enable VT-d."

psairam commented on Documentation/5.4.1/BEFORE_5.4.1u4/README.rst:
```diff
@@ -0,0 +1,1762 @@
+
+.. Note:: Above kernel arguments are consumed by the another env file which include in deployment command `/usr/share/openstack-tripleo-heat-templates/environments/host-config-and-reboot.yaml`
+
+2. Use the `create_compute_avrs_role.sh <../../scripts/create_roles/create_compute_avrs_role.sh>`_ to create a roles file called ``compute-avrs-role.yaml``.
```
path need to be modified to "../../nuage-tripleo-heat-templates/scripts/create_roles/create_compute_avrs_role.sh"

psairam commented on Documentation/5.4.1/BEFORE_5.4.1u4/README.rst:
```diff
@@ -0,0 +1,1762 @@
+     cd /home/stack/nuage-tripleo-heat-templates/scripts/create_roles
+    ./create_compute_avrs_role.sh
+
+Above command will create a new ``ComputeAvrs``  role for your deployment, and compare it with sample compute-avrs-role-sample.yaml provided at https://github.com/nuagenetworks/nuage-ospdirector/tree/OSPD13/nuage-tripleo-heat-templates/roles/compute-avrs-role-sample.yaml .
```
provide a relative path instead of absolute

psairam commented on Documentation/5.4.1/README.rst:
```diff
@@ -273,54 +263,107 @@ Follow the steps in https://access.redhat.com/documentation/en-us/red_hat_openst
+    **For multi-role AVRS deployment** environment file can be found at : `compute-avrs-mutlirole-environment.yaml <../../nuage-tripleo-heat-templates/environments/compute-avrs-mutlirole-environment.yaml>`_ file
+
+
+    a. For AVRS deployment we need to configure  `/etc/fast-path.env`, we set the following parameters :
```
same as above

psairam replied:
is it resolved? I don't see the changes

psairam commented on Documentation/5.4.1/README.rst:
```diff
@@ -273,54 +263,107 @@ Follow the steps in https://access.redhat.com/documentation/en-us/red_hat_openst
+        FastPathOptions        =====>    FP_OPTIONS
+
+
+    b. For AVRS deployment we need to configure `nova.conf`, we set the following parameters :
```
same as above

psairam replied:
is it resolved? I don't see the changes

psairam commented on Documentation/5.4.1/README.rst:
```diff
@@ -273,54 +263,107 @@ Follow the steps in https://access.redhat.com/documentation/en-us/red_hat_openst
+              monkey_patch_modules:
+                 value: nova.virt.libvirt.vif:openstack_6wind_extensions.queens.nova.virt.libvirt.vif.decorator
+
+    c. For AVRS deployment we need to configure kernel arguments we set following parameters:
```
same as above

psairam replied:
is it resolved? I don't see the changes

psairam commented on Documentation/5.4.1/README.rst:
```diff
@@ -664,238 +745,65 @@ In OSPD 13 and later, /usr/share/openstack-tripleo-heat-templates/environments/n
+    release: 5-4-1-u4
+    #Tag for Nuage container images
+    tag: latest
+    #Undercloud Local Registry IP Address : PORT
```
remove space after Address and before PORT. Also change PORT to Port

psairam commented on nuage-tripleo-heat-templates/scripts/pull_nuage_containers/nuage_overcloud_images.yaml.j2:
```diff
@@ -0,0 +1,13 @@
+  DockerInsecureRegistryAddress:
+  - {{ nuage_config['local_registry'] }}
+  DockerNeutronApiImage: {{ nuage_config['local_registry'] }}/nuagenetworks/rhosp{{ nuage_config['version'] }}-openstack-neutron-server-{{ nuage_config['release'] }}:{{ nuage_config['tag'] }}
+  DockerNeutronConfigImage: {{ nuage_config['local_registry'] }}/nuagenetworks/rhosp{{ nuage_config['version'] }}-openstack-neutron-server-{{ nuage_config['release'] }}:{{ nuage_config['tag'] }}
```
we need to add nova-compute images for AVRS
```
  DockerNovaComputeAvrsImage: {{ nuage_config['local_registry'] }}/nuagenetworks/rhosp{{ nuage_config['version'] }}-nova-compute-{{ nuage_config['release'] }}:{{ nuage_config['tag'] }}
  DockerNovaLibvirtAvrsConfigImage: {{ nuage_config['local_registry'] }}/nuagenetworks/rhosp{{ nuage_config['version'] }}-nova-compute-{{ nuage_config['release'] }}:{{ nuage_config['tag'] }}
```

psairam commented on Documentation/5.4.1/README.rst:
```diff
@@ -165,19 +156,18 @@ The integration includes the following steps:
-    - Apply the changes in the diff files in https://github.com/nuagenetworks/nuage-ospdirector/tree/OSPD13/tripleo-heat-templates-diff to the Undercloud codebase.
-    - The instructions to apply the patch script are in this README file: https://github.com/nuagenetworks/nuage-ospdirector/tree/OSPD13/tripleo-heat-templates-diff/README.md .
-    - For AVRS integration, get the script and files to patch the Overcloud image with the AVRS RPMs.
+    - Nuage provides heat templates & environment files to configure neutron on controller and nuage-openvswitch & nuage-metadata-agent on compute nodes.
```
Just a small thought, will it be nice if we mention that we configure VA on computeavrs nodes?

psairam commented on Documentation/5.4.1/README.rst:
```diff
@@ -165,19 +156,18 @@ The integration includes the following steps:
+    - For AVRS intergration, we can also create Multiple roles which we can pass different sets of configuration on those AVRS Compute Node. Sample enviroment file can be found `here <../../nuage-tripleo-heat-templates/environments/compute-avrs-multirole-environment.yaml>`_
 
 
 * Updating the Docker Images
```
This is no longer valid, please remove this section.

psairam commented on Documentation/5.4.1/README.rst:
```diff
@@ -200,13 +190,14 @@ Links to Nuage and OpenStack Resources
 * For the Nuage and Puppet modules, go to http://git.openstack.org/cgit/openstack/tripleo-heat-templates/tree/puppet .
 * For the files and script to generate the CMS ID, go to https://github.com/nuagenetworks/nuage-ospdirector/tree/OSPD13/generate-cms-id .
 
+.. Note::  **Please contact Nuage for Nuage Ironic Integration**
```
how about moving this to Readme of this repo itself.

psairam commented on Documentation/5.4.1/README.rst:
```diff
@@ -273,54 +263,107 @@ Follow the steps in https://access.redhat.com/documentation/en-us/red_hat_openst
-    FastPathOptions        =====>    FP_OPTIONS
+**For an AVRS integration please follow below steps as well**:
+
+1.  User can have one role for AVRS or Mutli-Roles for AVRS nodes.
```
replace one with single

psairam commented on Documentation/5.4.1/README.rst:
```diff
@@ -273,54 +263,107 @@ Follow the steps in https://access.redhat.com/documentation/en-us/red_hat_openst
-Above command will create a new ``ComputeAvrs``  role for your deployment, and compare it with sample avrs-role.yaml provided at https://github.com/nuagenetworks/nuage-ospdirector/tree/OSPD13/avrs/avrs-role.yaml .
-(Please Note, given ``avrs-role.yaml`` file can get updated with newer release )
+
+Above command will create a new ``ComputeAvrs``  role for your deployment, and compare it with sample `compute-avrs-role-sample.yaml <../../nuage-tripleo-heat-templates/roles/compute-avrs-role-sample.yaml>`_ .
```
lets move below note to here?

psairam commented on Documentation/5.4.1/README.rst:
```diff
@@ -426,6 +493,20 @@ For AVRS integration, follow these steps:
     OvercloudComputeAvrsFlavor: computeavrs
     ComputeAvrsCount: 2
 
+**For multi-role AVRS deployment**, set the `node-info.yaml` with the corresponding role name. The following example shows how to create a deployment with one Controller node, two Compute nodes,  two ComputeAvrsSingle & two ComputeAvrsDual Avrs nodes:
```
we can remove Avrs at last

psairam commented on Documentation/5.4.1/README.rst:
```diff
@@ -426,6 +493,20 @@ For AVRS integration, follow these steps:
+    ComputeCount: 2
+    OvercloudComputeAvrsSingleFlavor: computeavrssignle
+    ComputeAvrsSingleCount: 2
+    OvercloudComputeAvrsSingleFlavor: computeavrsdual
```
name should be "OvercloudComputeAvrsDualFlavor"

psairam commented on Documentation/5.4.1/README.rst:
```diff
@@ -273,54 +263,108 @@ Follow the steps in https://access.redhat.com/documentation/en-us/red_hat_openst
+        FastPathOptions        =====>    FP_OPTIONS
+
+
+    b. For AVRS deployment s requires to configure monkey_patch parameters in `nova.conf` and we use below to configure them.
```
lets say, "AVRS deployment" and remove extra "s"

psairam commented on Documentation/5.4.1/README.rst:
```diff
@@ -273,54 +263,108 @@ Follow the steps in https://access.redhat.com/documentation/en-us/red_hat_openst
+
+        KernelArgs: "hugepages=12831 iommu=pt intel_iommu=on isolcpus=1-7"
+
+    .. Note:: Above kernel arguments are consumed by the another env file which include in deployment command `/usr/share/openstack-tripleo-heat-templates/environments/host-config-and-reboot.yaml`
```
actually, it is heat template that consumes the parameters and here it is `/usr/share/openstack-tripleo-heat-templates/extraconfig/pre_network/host_config_and_reboot.yaml`

psairam commented on Documentation/5.4.1/README.rst:
```diff
@@ -664,238 +746,65 @@ In OSPD 13 and later, /usr/share/openstack-tripleo-heat-templates/environments/n
-      ...
-      -e /usr/share/openstack-tripleo-heat-templates/environments/services-docker/ironic.yaml \
-      ...
+**Nuage containers from Redhat Partner Container Catalog (For Nuage release greater than or equals 5.4.1u4)**
```
this doc itself is for releases >= 5.4.1.U4 right, so do we need to mention it explicitly

psairam commented on Documentation/5.4.1/README.rst:
```diff
@@ -912,65 +821,58 @@ Use the ``openstack overcloud deploy`` command options to pass the environment f
-    openstack overcloud deploy --templates --libvirt-type qemu -e /home/stack/templates/overcloud_images.yaml -e /home/stack/templates/node-info.yaml -e /home/stack/templates/docker-insecure-registry.yaml -e /usr/share/openstack-tripleo-heat-templates/environments/neutron-nuage-config.yaml -e /usr/share/openstack-tripleo-heat-templates/environments/nova-nuage-config.yaml --ntp-server ntp-server
+    openstack overcloud deploy --templates --libvirt-type qemu -e /home/stack/templates/overcloud_images.yaml -e /home/stack/templates/node-info.yaml -e /home/stack/templates/docker-insecure-registry.yaml -e /home/stack/nuage-tripleo-heat-templates/environments/neutron-nuage-config.yaml -e /home/stack/nuage-tripleo-heat-templates/environments/nova-nuage-config.yaml --ntp-server ntp-server
     
     For AVRS integration, use:
```
we need to add deploy commands for multi-role avrs deployments

psairam commented on Documentation/5.4.1/README.rst:
```diff
@@ -912,65 +821,58 @@ Use the ``openstack overcloud deploy`` command options to pass the environment f
-    openstack overcloud deploy --templates --libvirt-type qemu -e /home/stack/templates/overcloud_images.yaml -e /home/stack/templates/node-info.yaml -e /home/stack/templates/docker-insecure-registry.yaml -e /usr/share/openstack-tripleo-heat-templates/environments/neutron-nuage-config.yaml -e /usr/share/openstack-tripleo-heat-templates/environments/nova-nuage-config.yaml --ntp-server ntp-server
+    openstack overcloud deploy --templates --libvirt-type qemu -e /home/stack/templates/overcloud_images.yaml -e /home/stack/templates/node-info.yaml -e /home/stack/templates/docker-insecure-registry.yaml -e /home/stack/nuage-tripleo-heat-templates/environments/neutron-nuage-config.yaml -e /home/stack/nuage-tripleo-heat-templates/environments/nova-nuage-config.yaml --ntp-server ntp-server
     
     For AVRS integration, use:
```
same as above

psairam commented on Documentation/5.4.1/README.rst:
```diff
@@ -912,65 +821,58 @@ Use the ``openstack overcloud deploy`` command options to pass the environment f
-    openstack overcloud deploy --templates -e /home/stack/templates/overcloud_images.yaml -e /home/stack/templates/docker-insecure-registry.yaml -e /usr/share/openstack-tripleo-heat-templates/environments/network-environment.yaml -e /home/stack/templates/node-info.yaml -e /usr/share/openstack-tripleo-heat-templates/environments/network-isolation.yaml -e /usr/share/openstack-tripleo-heat-templates ...
+    openstack overcloud deploy --templates -e /home/stack/templates/overcloud_images.yaml -e /home/stack/templates/docker-insecure-registry.yaml -e /usr/share/openstack-tripleo-heat-templates/environments/network-environment.yaml -e /home/stack/templates/node-info.yaml -e /usr/share/openstack-tripleo-heat-templates/environments/network-isolation.yaml -e /usr/share/openstack-tripleo-heat-templates ...
 
     For AVRS integration, use the following:
```
same as above

psairam commented on Documentation/5.4.1/README.rst:
```diff
@@ -981,10 +883,9 @@ where:
    * ``neutron-sriov.yaml`` Neutron SRIOV specific parameter values
-   * ``avrs-role.yaml`` Enables services required for Compute Avrs role
-   * ``ironic-role.yaml`` Enables Ironic Inspector service for Controller role
+   * ``compute-avrs-role.yaml`` Enables services required for Compute Avrs role
```
after resolving above 3 comments, add compute-avrs-multi-role.yaml here

psairam commented on Documentation/5.4.1/README.rst:
```diff
@@ -1506,66 +1374,74 @@ nova-nuage-config.yaml For a KVM Setup
-      # A list or range of physical CPU cores to be tuned.
-      # The given args will be appended to the tuned cpu-partitioning profile.
-      HostIsolatedCoreList: "1-7,9-15"
       FastPathNics: "0000:06:00.1 0000:06:00.2"
```
it is optional, let's move FP params inside "ComputeAvrsParameters"?

psairam commented on Documentation/5.4.1/README.rst:
```diff
@@ -1506,66 +1374,74 @@ nova-nuage-config.yaml For a KVM Setup
+      OS::TripleO::Services::NovaComputeAvrs: ../docker/services/nova-compute-avrs.yaml
 
     parameter_defaults:
       NovaVcpuPinSet: "2-7,10-15"
```
this need to be moved under "ComputeAvrsParameters"

psairam commented on Documentation/5.4.1/README_U1_to_U3.rst:
```diff
@@ -0,0 +1,1800 @@
+.. ..  =========  =======    =========
+.. ..  Date       Version    Author
+.. ..  =========  =======    =========
+.. ..  05/24/18    5.3.2     Jennifer - Create new document for Queens ML2 based on the 5.3.1 Queens files.
```
do we need to have 5.3.3 info here or can this removed in the same way as 5.4.1/README.rst

psairam commented on Documentation/5.4.1/README_U1_to_U3.rst:
```diff
@@ -0,0 +1,1800 @@
+    FastPathOptions        =====>    FP_OPTIONS
+
+
+b. Avrs requires to configure monkey_patch parameters in nova.conf and we use below to configure them.
```
this point looks good, and same can be used in 5.4.1.U4 readme

psairam commented on Documentation/5.4.1/README_U1_to_U3.rst:
```diff
@@ -0,0 +1,1800 @@
+
+    KernelArgs: "hugepages=12831 iommu=pt intel_iommu=on isolcpus=1-7"
+
+.. Note:: Above kernel arguments are consumed by the another env file which include in deployment command `/usr/share/openstack-tripleo-heat-templates/environments/host-config-and-reboot.yaml`
```
same as 5.4.1U4 doc

psairam commented on Documentation/5.4.1/README_U1_to_U3.rst:
```diff
@@ -0,0 +1,1800 @@
+
+.. Note:: Above kernel arguments are consumed by the another env file which include in deployment command `/usr/share/openstack-tripleo-heat-templates/environments/host-config-and-reboot.yaml`
+
+2. Use the `create_compute_avrs_role.sh <../../nuage-tripleo-heat-templates/scripts/create_roles/create_compute_avrs_role.sh>`_ to create a roles file called ``compute-avrs-role.yaml``.
```
lets swap 1 and 2. So, we talk about role creation first and env file later. It will be in the same way as 5.4.1U4 doc

psairam commented on nuage-tripleo-heat-templates/scripts/create_roles/create_compute_avrs_multirole.sh:
```diff
@@ -0,0 +1,41 @@
+#!/usr/bin/env bash
```
please check the mode on this file, because when I ran "./create_compute_avrs_multirole.sh" I got "Permission denied"

psairam commented on nuage-tripleo-heat-templates/scripts/create_roles/create_compute_avrs_role.sh:
```diff
@@ -0,0 +1,37 @@
+#!/usr/bin/env bash
```
please check the mode on this file, because when I ran "./create_compute_avrs_role.sh" I got "Permission denied"

psairam commented on nuage-tripleo-heat-templates/environments/compute-avrs-environment.yaml:
```diff
@@ -0,0 +1,23 @@
+    NovaVcpuPinSet: "2-7,10-15"
+    # An array of filters used by Nova to filter a node.These filters will be applied in the order they are listed,
+    # so place your most restrictive filters first to make the filtering process more efficient.
+    NovaSchedulerDefaultFilters: "RetryFilter,AvailabilityZoneFilter,RamFilter,ComputeFilter,ComputeCapabilitiesFilter,ImagePropertiesFilter,ServerGroupAntiAffinityFilter,ServerGroupAffinityFilter,PciPassthroughFilter,NUMATopologyFilter,AggregateInstanceExtraSpecsFilter"
```
this should be out of "ComputeAvrsParameters:", because it gets configured on controller node but not on computeavrs

psairam commented on nuage-tripleo-heat-templates/scripts/create_roles/create_compute_avrs_role.sh:
```diff
@@ -0,0 +1,37 @@
+sudo sed -i -e 's/ Compute/ ComputeAvrs/g' ../../roles/ComputeAvrs.yaml
+sudo sed -i -e "s/HostnameFormatDefault: '%stackname%-compute-%index%'/HostnameFormatDefault: '%stackname%-computeavrs-%index%'/g" ../../roles/ComputeAvrs.yaml
+sudo sed -i -e 's/- OS::TripleO::Services::NovaCompute/- OS::TripleO::Services::NovaComputeAvrs/g'   ../../roles/ComputeAvrs.yaml
+openstack overcloud roles generate --roles-path ../../roles -o ../../roles/compute-avrs-role.yaml Controller Compute ComputeAvrs
```
lets only generate ComputeAvrs.yaml but not "compute-avrs-role.yaml", because if we see for sriov or ironic we are asking users to generate the roles file manually. Moreover, it will be a exercise for users on how to generate a role-file with ComputeAvrs role if there are planning to deploy Compute + ComputeAvrs + ComputeSriov + Ironic.

psairam commented on nuage-tripleo-heat-templates/scripts/create_roles/create_compute_avrs_multirole.sh:
```diff
@@ -0,0 +1,41 @@
+done
+
+
+openstack overcloud roles generate --roles-path ../../roles -o ../../roles/compute-avrs-role.yaml Controller Compute  ${roles[*]}
```
lets only generate ${role}.yaml but not "compute-avrs-role.yaml", because if we see for sriov or ironic we are asking users to generate the roles file manually. Moreover, it will be a exercise for users on how to generate a role-file with ${role} role if there are planning to deploy Compute + ${role} + ComputeSriov + Ironic.

psairam commented on Documentation/5.4.1/README.rst:
```diff
@@ -438,11 +520,11 @@ For AVRS integration, follow these steps:
 ::
 
-    openstack overcloud roles generate Controller Compute ComputeSriov -o /home/stack/templates/sriov-role.yaml
+    openstack overcloud roles generate Controller Compute ComputeSriov -o /home/stack/nuage-tripleo-heat-templates/roles/compute-sriov-role.yaml
```
Here we need to do 2 things, we have to make a copy of all roles from /usr/share/tht/roles to nuage-tripleo-heat-templates/roles/.
Then generate the role file by passing "--roles-path /home/stack/nuage-tripleo-heat-templates/roles/" and generate the role-file in /home/stack/templates/.
Because ideally /home/stack/nuage-tripleo-heat-templates/roles/ should only have all available roles but not role-files.

psairam commented on Documentation/5.4.1/README_U1_to_U3.rst:
```diff
@@ -0,0 +1,1800 @@
+
+::
+
+    openstack overcloud roles generate Controller Compute ComputeSriov -o /home/stack/nuage-tripleo-heat-templates/roles/compute-sriov-role.yaml
```
same as in 5.4.1U4

psairam commented on Documentation/5.4.1/README_U1_to_U3.rst:
```diff
@@ -0,0 +1,1800 @@
+
+    ::
+
+        openstack overcloud roles generate Controller Compute -o /home/stack/nuage-tripleo-heat-templates/roles/ironic-role.yaml
```
same as in sriov step in 5.4.1U4

psairam commented on Documentation/5.4.1/README_U1_to_U3.rst:
```diff
@@ -0,0 +1,1800 @@
+      # override these by including one of the net-*.yaml environment files,
+      # such as net-bond-with-vlans.yaml, or modifying the list here.
+    {%- for role in roles %}
+      # Port assignment
```
same as in 5.4.1U4 doc

psairam commented on image-patching/stopgap-script/nuage_overcloud_full_patch.py:
```diff
@@ -104,37 +101,28 @@ def virt_copy(command):
+        subscription_command = subscription_command + 'cat <<EOT > rhel_subscription \n'
+
+    enable_pool = 'subscription-manager register --username=%s --password=\'%s\' \n' \
+          'subscription-manager subscribe --pool=%s \n' \
```
lets replace "subscribe" with "attach" as it is deprecated.

psairam commented on nuage-tripleo-heat-templates/scripts/pull_nuage_containers/nuage_overcloud_images.yaml.j2:
```diff
@@ -0,0 +1,17 @@
+  - {{ nuage_config['local_registry'] }}
+  DockerNeutronApiImage: {{ nuage_config['local_registry'] }}/nuagenetworks/rhosp{{ nuage_config['version'] }}-openstack-neutron-server-{{ nuage_config['release'] }}:{{ nuage_config['tag'] }}
+  DockerNeutronConfigImage: {{ nuage_config['local_registry'] }}/nuagenetworks/rhosp{{ nuage_config['version'] }}-openstack-neutron-server-{{ nuage_config['release'] }}:{{ nuage_config['tag'] }}
+  {% if 'nova-compute' in nuage_config['nuage_images' %}
```
its ok to have extra images in this env file. Unless docker/service/nova-compute-avrs.yaml is called in a deployment, these won't come into picture at all. So, we can remove this condition.

psairam replied:
Error: jinja2.exceptions.TemplateSyntaxError: unexpected '}', expected ']'
Add ']' after "/'nuage_images/'"

psairam commented on nuage-tripleo-heat-templates/scripts/pull_nuage_containers/nuage_container_config.yaml:
```diff
@@ -0,0 +1,10 @@
+#Undercloud Local Registry IP Address : PORT
+local_registry: 192.168.24.1:8787
+#List of Nuage containers ( For AVRS, please include 'nova-compute' as well)
+nuage_images: ['heat-api-cfn', 'heat-api', 'heat-engine', 'horizon', 'neutron-server']
```
include nova-compute by default

psairam commented on nuage-tripleo-heat-templates/scripts/pull_nuage_containers/nuage_container_config.yaml:
```diff
@@ -0,0 +1,10 @@
+release: 5-4-1-u3
+#Tag for Nuage container images
+tag: latest
+#Undercloud Local Registry IP Address : PORT
```
no space need before and after ":"

psairam commented on Documentation/5.4.1/README.rst:
```diff
@@ -664,238 +746,65 @@ In OSPD 13 and later, /usr/share/openstack-tripleo-heat-templates/environments/n
+    nuage_images: ['heat-api-cfn', 'heat-api', 'heat-engine', 'horizon', 'neutron-server', 'nova-compute']
 
-8. For all the Docker files in the Nuage-OSPD-Dockerfiles directory, change the ``<tag>`` of the Docker base image to point to the same tag in ``/home/stack/templates/overcloud_images.yaml`` .
+4. Now execute `nuage_containers_pull.py` script by passing nuage_config.yaml to "--nuage-config" argument.
```
nuage_config.yaml was renamed to  nuage_container_config.yaml

psairam replied:
nuage_containers_pull.py is renamed to nuage_container_pull.py

psairam commented on Documentation/5.4.1/README.rst:
```diff
@@ -664,238 +746,65 @@ In OSPD 13 and later, /usr/share/openstack-tripleo-heat-templates/environments/n
-    FROM <docker-image-name>:<tag>
-    Example:
-    FROM registry.access.redhat.com/rhosp13/openstack-neutron-server:13.0-60.1543534138
+    $ python nuage_containers_pull.py --nuage-config nuage_config.yaml
```
nuage_config.yaml was renamed to nuage_container_config.yaml

psairam replied:
nuage_containers_pull.py is renamed to nuage_container_pull.py

psairam commented on nuage-tripleo-heat-templates/scripts/create_roles/create_compute_avrs_multirole.sh:
```diff
@@ -0,0 +1,37 @@
+    sudo openstack overcloud roles generate --roles-path ../../roles -o ../../roles/${role}.yaml Compute
+    sudo sed -i -e "s/ Compute/ ${role}/g" ../../roles/${role}.yaml
+    sudo sed -i -e "s/HostnameFormatDefault: '%stackname%-compute-%index%'/HostnameFormatDefault: '%stackname%-${role,,}-%index%'/g" ../../roles/${role}.yaml
+    sudo sed -i -e "s/- OS::TripleO::Services::NovaCompute/- OS::TripleO::Services::${role}/g"   ../../roles/${role}.yaml
```
In nuage-tripleo-heat-templates/environments/compute-avrs-multirole-environment.yaml we are using OS::TripleO::Services::NovaComputeAvrs, so lets keep this service name common across all multiple avrs roles.

psairam commented on nuage-tripleo-heat-templates/docker/services/nova-compute-avrs.yaml:
```diff
@@ -0,0 +1,300 @@
+        config_image: {get_param: DockerNovaLibvirtAvrsConfigImage}
+      kolla_config:
+        get_attr: [NovaComputeAvrsDockerBase, role_data, kolla_config]
+      docker_config_scripts:Making Re
```
this line has some unnecessary data after ":", please remove it

psairam commented on nuage-tripleo-heat-templates/environments/compute-avrs-multirole-environment.yaml:
```diff
@@ -0,0 +1,35 @@
+         value: true
+      DEFAULT/monkey_patch_modules:
+         value: nova.virt.libvirt.vif:openstack_6wind_extensions.queens.nova.virt.libvirt.vif.decorator
+  NovaComputeAvrsSingleParameters:
```
role name doesn't have Nova in it

psairam commented on nuage-tripleo-heat-templates/environments/compute-avrs-multirole-environment.yaml:
```diff
@@ -0,0 +1,35 @@
+    FastPathNics: "0000:06:00.1 0000:06:00.2"
+    FastPathMask: "1,9"
+    FastPathNicDescriptors: "--nb-rxd=4096 --nb-txd=4096"
+    FastPathOptions: "\"--mod-opt=fp-vswitch:--flows=200000 --max-nfct=40000\""
```
role name doesn't have Nova in it

psairam commented on Documentation/5.4.1/README.rst:
```diff
@@ -912,65 +821,58 @@ Use the ``openstack overcloud deploy`` command options to pass the environment f
-::
-
-    openstack overcloud deploy --templates -r /home/stack/templates/ironic-role.yaml -e /home/stack/templates/overcloud_images.yaml -e /home/stack/templates/node-info.yaml -e /home/stack/templates/docker-insecure-registry.yaml -e /usr/share/openstack-tripleo-heat-templates/environments/neutron-nuage-config.yaml -e /usr/share/openstack-tripleo-heat-templates/environments/nova-nuage-config.yaml -e  ...
+    openstack overcloud deploy --templates -r /home/stack/nuage-tripleo-heat-templates/roles/compute-avrs-role.yaml -e /home/stack/templates/overcloud_images.yaml -e /home/stack/templates/docker-insecure-registry.yaml -e /usr/share/openstack-tripleo-heat-templates/environments/network-environment.yaml -e /home/stack/templates/node-info.yaml -e /usr/share/openstack-tripleo-heat-templates/environme ...
```
lets remove `/home/stack/templates/docker-insecure-registry.yaml`, because 
1) insecure registry is laready being set in `/home/stack/nuage-tripleo-heat-templates/environments/nuage_overcloud_images.yaml`
2) we don't talk about creating this file anymore in our documentation.

psairam replied:
add `/home/stack/nuage-tripleo-heat-templates/environments/nuage_overcloud_images.yaml` as env file

psairam commented on Documentation/5.4.1/README.rst:
```diff
@@ -912,65 +821,58 @@ Use the ``openstack overcloud deploy`` command options to pass the environment f
     
     For a virtual deployment, add the --libvirt-type parameter:
-    openstack overcloud deploy --templates --libvirt-type qemu -e /home/stack/templates/overcloud_images.yaml -e /home/stack/templates/node-info.yaml -e /home/stack/templates/docker-insecure-registry.yaml -e /usr/share/openstack-tripleo-heat-templates/environments/neutron-nuage-config.yaml -e /usr/share/openstack-tripleo-heat-templates/environments/nova-nuage-config.yaml --ntp-server ntp-server
+    openstack overcloud deploy --templates --libvirt-type qemu -e /home/stack/templates/overcloud_images.yaml -e /home/stack/templates/node-info.yaml -e /home/stack/templates/docker-insecure-registry.yaml -e /home/stack/nuage-tripleo-heat-templates/environments/neutron-nuage-config.yaml -e /home/stack/nuage-tripleo-heat-templates/environments/nova-nuage-config.yaml --ntp-server ntp-server
```
add `/home/stack/nuage-tripleo-heat-templates/environments/nuage_overcloud_images.yaml` as env file

psairam commented on Documentation/5.4.1/README.rst:
```diff
@@ -912,65 +821,58 @@ Use the ``openstack overcloud deploy`` command options to pass the environment f
     
     For AVRS integration, use:
-    openstack overcloud deploy --templates -r /home/stack/templates/avrs-role.yaml -e /home/stack/templates/overcloud_images.yaml -e /home/stack/templates/node-info.yaml -e /home/stack/templates/docker-insecure-registry.yaml  -e /usr/share/openstack-tripleo-heat-templates/environments/nova-nuage-config.yaml -e /usr/share/openstack-tripleo-heat-templates/environments/neutron-nuage-config.yaml -e / ...
+    openstack overcloud deploy --templates -r /home/stack/nuage-tripleo-heat-templates/roles/compute-avrs-role.yaml -e /home/stack/templates/overcloud_images.yaml -e /home/stack/templates/node-info.yaml -e /home/stack/templates/docker-insecure-registry.yaml  -e /home/stack/nuage-tripleo-heat-templates/environments/nova-nuage-config.yaml -e /home/stack/nuage-tripleo-heat-templates/environments/neu ...
```
add `/home/stack/nuage-tripleo-heat-templates/environments/nuage_overcloud_images.yaml` as env file

psairam commented on Documentation/5.4.1/README.rst:
```diff
@@ -912,65 +821,58 @@ Use the ``openstack overcloud deploy`` command options to pass the environment f
 ::
 
-    openstack overcloud deploy --templates -e /home/stack/templates/overcloud_images.yaml -e /home/stack/templates/node-info.yaml -e /home/stack/templates/docker-insecure-registry.yaml -e /usr/share/openstack-tripleo-heat-templates/environments/neutron-nuage-config.yaml -e /usr/share/openstack-tripleo-heat-templates/environments/nova-nuage-config.yaml --ntp-server ntp-server
+    openstack overcloud deploy --templates -e /home/stack/templates/overcloud_images.yaml -e /home/stack/templates/node-info.yaml -e /home/stack/templates/docker-insecure-registry.yaml -e /home/stack/nuage-tripleo-heat-templates/environments/neutron-nuage-config.yaml -e /home/stack/nuage-tripleo-heat-templates/environments/nova-nuage-config.yaml --ntp-server ntp-server
```
add `/home/stack/nuage-tripleo-heat-templates/environments/nuage_overcloud_images.yaml` as env file

psairam commented on Documentation/5.4.1/README.rst:
```diff
@@ -912,65 +821,58 @@ Use the ``openstack overcloud deploy`` command options to pass the environment f
     
     For a virtual deployment, add the --libvirt-type parameter:
-    openstack overcloud deploy --templates --libvirt-type qemu -e /home/stack/templates/overcloud_images.yaml -e /home/stack/templates/node-info.yaml -e /home/stack/templates/docker-insecure-registry.yaml -e /usr/share/openstack-tripleo-heat-templates/environments/neutron-nuage-config.yaml -e /usr/share/openstack-tripleo-heat-templates/environments/nova-nuage-config.yaml --ntp-server ntp-server
+    openstack overcloud deploy --templates --libvirt-type qemu -e /home/stack/templates/overcloud_images.yaml -e /home/stack/templates/node-info.yaml -e /home/stack/templates/docker-insecure-registry.yaml -e /home/stack/nuage-tripleo-heat-templates/environments/neutron-nuage-config.yaml -e /home/stack/nuage-tripleo-heat-templates/environments/nova-nuage-config.yaml --ntp-server ntp-server
```
add `/home/stack/nuage-tripleo-heat-templates/environments/nuage_overcloud_images.yaml` as env file

psairam commented on Documentation/5.4.1/README.rst:
```diff
@@ -912,65 +821,58 @@ Use the ``openstack overcloud deploy`` command options to pass the environment f
     
     For AVRS integration, use:
-    openstack overcloud deploy --templates -r /home/stack/templates/avrs-role.yaml -e /home/stack/templates/overcloud_images.yaml -e /home/stack/templates/node-info.yaml -e /home/stack/templates/docker-insecure-registry.yaml  -e /usr/share/openstack-tripleo-heat-templates/environments/nova-nuage-config.yaml -e /usr/share/openstack-tripleo-heat-templates/environments/neutron-nuage-config.yaml -e / ...
+    openstack overcloud deploy --templates -r /home/stack/nuage-tripleo-heat-templates/roles/compute-avrs-role.yaml -e /home/stack/templates/overcloud_images.yaml -e /home/stack/templates/node-info.yaml -e /home/stack/templates/docker-insecure-registry.yaml  -e /home/stack/nuage-tripleo-heat-templates/environments/nova-nuage-config.yaml -e /home/stack/nuage-tripleo-heat-templates/environments/neu ...
```
add `/home/stack/nuage-tripleo-heat-templates/environments/nuage_overcloud_images.yaml` as env file

psairam commented on Documentation/5.4.1/README.rst:
```diff
@@ -912,65 +821,58 @@ Use the ``openstack overcloud deploy`` command options to pass the environment f
 ::
 
-   openstack overcloud deploy --templates -r /home/stack/templates/sriov-role.yaml -e /home/stack/templates/overcloud_images.yaml -e /home/stack/templates/docker-insecure-registry.yaml -e /home/stack/templates/node-info.yaml -e /home/stack/templates/neutron-sriov.yaml -e /usr/share/openstack-tripleo-heat-templates/environments/neutron-nuage-config.yaml -e /usr/share/openstack-tripleo-heat-templat ...
+   openstack overcloud deploy --templates -r /home/stack/nuage-tripleo-heat-templates/roles/compute-sriov-role.yaml -e /home/stack/templates/overcloud_images.yaml -e /home/stack/templates/docker-insecure-registry.yaml -e /home/stack/templates/node-info.yaml -e /home/stack/templates/neutron-sriov.yaml -e /home/stack/nuage-tripleo-heat-templates/environments/neutron-nuage-config.yaml -e /home/stack ...
```
add `/home/stack/nuage-tripleo-heat-templates/environments/nuage_overcloud_images.yaml` as env file

psairam commented on Documentation/5.4.1/README.rst:
```diff
@@ -912,65 +821,58 @@ Use the ``openstack overcloud deploy`` command options to pass the environment f
 ::
 
-    openstack overcloud deploy --templates -e /home/stack/templates/overcloud_images.yaml -e /home/stack/templates/docker-insecure-registry.yaml -e /usr/share/openstack-tripleo-heat-templates/environments/network-environment.yaml -e /home/stack/templates/node-info.yaml -e /usr/share/openstack-tripleo-heat-templates/environments/network-isolation.yaml -e /usr/share/openstack-tripleo-heat-templates ...
+    openstack overcloud deploy --templates -e /home/stack/templates/overcloud_images.yaml -e /home/stack/templates/docker-insecure-registry.yaml -e /usr/share/openstack-tripleo-heat-templates/environments/network-environment.yaml -e /home/stack/templates/node-info.yaml -e /usr/share/openstack-tripleo-heat-templates/environments/network-isolation.yaml -e /usr/share/openstack-tripleo-heat-templates ...
```
add `/home/stack/nuage-tripleo-heat-templates/environments/nuage_overcloud_images.yaml` as env file

psairam commented on nuage-tripleo-heat-templates/scripts/pull_nuage_containers/nuage_overcloud_images.yaml.j2:
```diff
@@ -0,0 +1,15 @@
+  - {{ nuage_config['local_registry'] }}
+  DockerNeutronApiImage: {{ nuage_config['local_registry'] }}/nuagenetworks/rhosp{{ nuage_config['version'] }}-openstack-neutron-server-{{ nuage_config['release'] }}:{{ nuage_config['tag'] }}
+  DockerNeutronConfigImage: {{ nuage_config['local_registry'] }}/nuagenetworks/rhosp{{ nuage_config['version'] }}-openstack-neutron-server-{{ nuage_config['release'] }}:{{ nuage_config['tag'] }}
+  DockerNovaComputeAvrsImage: {{ nuage_config['local_registry'] }}/nuagenetworks/rhosp{{ nuage_config['version'] }}-nova-compute-{{ nuage_config['release'] }}:{{ nuage_config['tag'] }}
```
"openstack-" is missing after "rhosp{{ nuage_config['version'] }}-"

sunny-verma committed:
Resolving Issue #69 and some readme changes

Change from --no_signing_key to  --no-signing-key

This commit also Document changes.

psairam committed:
refactoring network templates

sunny-verma committed:
Adding support for relative and absolute path for GPG key

psairam commented on Documentation/5.4.1/README.rst:
```diff
@@ -273,54 +252,67 @@ Follow the steps in https://access.redhat.com/documentation/en-us/red_hat_openst
-::
+    Above command will create a new ``ComputeAvrs``  role for your deployment, and compare it with sample `compute-avrs-role-sample.yaml <../../nuage-tripleo-heat-templates/templates/compute-avrs-role-sample.yaml>`_ .
+
+    **For mutli-role AVRS deployment**, we have automated `script <../../nuage-tripleo-heat-templates/scripts/create_compute_avrs_multirole.sh>`_ to create ComputeAvrsSingle & ComputeAvrsDual role. You can edit these files with your requirements to create new roles.
```
correct relative path is "nuage-tripleo-heat-templates/scripts/create_roles/create_compute_avrs_multirole.sh"

psairam commented on Documentation/5.4.1/README.rst:
```diff
@@ -375,7 +367,15 @@ Phase 7: Create the Heat templates.
     * neutron-nuage-config.yaml - Add the generated ``cms_id`` to the ``NeutronNuageCMSId`` parameter.
```
In point 1, change the path from /usr/share/openstack-tripleo-heat-templates/environments/ to /home/stack/nuage-tripleo-heat-templates/environments/

psairam commented on Documentation/5.4.1/README.rst:
```diff
@@ -1506,75 +1303,74 @@ nova-nuage-config.yaml For a KVM Setup
 
-avrs-environment.yaml for AVRS integration
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+compute-avrs-environment.yaml for AVRS integration
```
In line 1206, replace OS::TripleO::Services::NeutronCorePluginML2Nuage to ../puppet/services/neutron-plugin-ml2-nuage.yaml

psairam replied:
In 1293, replace OS::TripleO::Services::ComputeNeutronCorePluginNuage with ../puppet/services/neutron-compute-plugin-nuage.yaml

psairam commented on Documentation/5.4.1/README_U1_to_U3.rst:
```diff
@@ -0,0 +1,1716 @@
+Phase 7: Create the Heat templates.
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+1. Go to /usr/share/openstack-tripleo-heat-templates/environments/ on the Undercloud machine.
```
change /usr/share/openstack-tripleo-heat-templates/environments/ to /home/stack/nuage-tripleo-heat-templates/environments/

psairam commented on Documentation/5.4.1/README_U1_to_U3.rst:
```diff
@@ -0,0 +1,1716 @@
+      OS::TripleO::Services::NeutronOvsAgent: OS::Heat::None
+      OS::TripleO::Services::ComputeNeutronOvsAgent: OS::Heat::None
+      # Override the NeutronCorePlugin to use Nuage
+      OS::TripleO::Docker::NeutronMl2PluginBase: OS::TripleO::Services::NeutronCorePluginML2Nuage
```
same as U4

psairam commented on Documentation/5.4.1/README_U1_to_U3.rst:
```diff
@@ -0,0 +1,1716 @@
+      NeutronNuageCMSId: 'a91a28b8-28de-436b-a665-6d08a9346464'
+      UseForwardedFor: true
+      NeutronPluginMl2PuppetTags: 'neutron_plugin_ml2,neutron_plugin_nuage'
+      NeutronServ
```
same as U4

sunny-verma committed:
Adding some document changes

* adding parameter_defaults to node-info.yaml
* Added host-reboot file in overcloud deploy command
* correcting the relative path to nuage-tht
* Fixing OPENSTACK-2619

psairam commented on image-patching/stopgap-script/nuage_overcloud_full_patch.py:
```diff
@@ -392,9 +367,6 @@ def image_patching(args):
-
     cmds_run(['echo "Done"'])
 
 
```
We are not using Version any more, so remove it from line 421

sunny-verma replied:
Will be fixed in next commit

psairam commented on Documentation/5.4.1/README.rst:
```diff
@@ -391,40 +391,143 @@ This example shows how to create a deployment with one Controller node and two C
+
+        ComputeAvrsExtraConfig:
+            nova::config::nova_config:
+              monkey_patch:
```
```
            "Error: Parameter name failed on Nova_config[monkey_patch]: Invalid value \"monkey_patch\". Valid values match /\\S+\\/\\S+/. at /etc/puppet/modules/nova/manifests/config.pp:40",
```
we need mention the section as well. The modified line will be
```
DEFAULT/monkey_patch:
  value: true
DEFAULT/monkey_patch_modules:
  value: nova.virt.libvirt.vif:openstack_6wind_extensions.queens.nova.virt.libvirt.vif.decorator
```

psairam commented on Documentation/5.4.1/README.rst:
```diff
@@ -1506,75 +1307,74 @@ nova-nuage-config.yaml For a KVM Setup
+        FastPathOffload: "off"
+      ComputeAvrsExtraConfig:
+        nova::config::nova_config:
+          monkey_patch:
```
```
            "Error: Parameter name failed on Nova_config[monkey_patch]: Invalid value \"monkey_patch\". Valid values match /\\S+\\/\\S+/. at /etc/puppet/modules/nova/manifests/config.pp:40",
```
we need mention the section as well. The modified line will be
```
DEFAULT/monkey_patch:
  value: true
DEFAULT/monkey_patch_modules:
  value: nova.virt.libvirt.vif:openstack_6wind_extensions.queens.nova.virt.libvirt.vif.decorator
```

psairam commented on nuage-tripleo-heat-templates/environments/compute-avrs-environment.yaml:
```diff
@@ -0,0 +1,23 @@
+
+  ComputeAvrsExtraConfig:
+    nova::config::nova_config:
+      monkey_patch:
```
```
            "Error: Parameter name failed on Nova_config[monkey_patch]: Invalid value \"monkey_patch\". Valid values match /\\S+\\/\\S+/. at /etc/puppet/modules/nova/manifests/config.pp:40",
```
we need mention the section as well. The modified line will be
```
DEFAULT/monkey_patch:
  value: true
DEFAULT/monkey_patch_modules:
  value: nova.virt.libvirt.vif:openstack_6wind_extensions.queens.nova.virt.libvirt.vif.decorator
```

psairam commented on image-patching/stopgap-script/nuage_overcloud_full_patch.py:
```diff
@@ -164,69 +152,57 @@ def uninstall_packages(image):
+        '\$missing_packages" \n'
+        '    exit 1 \n'
+        'fi\n'
+        'EOT'.format(NUAGE_DEPENDENCIES, NUAGE_PACKAGES, NUAGE_VRS_PACKAGE)
```
NUAGE_PACKAGES has nuage-metadata-agent, which requires nuage-openvswitch and according to this fucntion, nuage-openvswitch is installing later in  NUAGE_VRS_PACKAGE
